pkg/solana/txm/txm.go

```diff
@@ -790,6 +813,41 @@ func (txm *Txm) Enqueue(ctx context.Context, accountID string, tx *solanaGo.Tran
 	return nil
 }
 
+func (txm *Txm) waitForTxStatus(ctx context.Context, transactionID string, desiredStatus types.TransactionStatus) error {
+	waitCtx, cancel := context.WithTimeout(ctx, txm.cfg.TxTimeout())
```

Comment:
Since the transaction is already broadcasted and we're just waiting for confirmation here, I think we should use `TxConfirmTimeout` instead.

Reply:
I actually changed it to wait for Finalization not confirmation, I figured the longer timeout may be safer since we don't have one specifically for finalization.

Reply:
Even though we don't have to worry about the context anymore, I think it might still be better to only wait for confirmed here. It should help throughput and tbh talking to one of the auditors it seems like re-orgs on confirmed tx never really happen since a super majority validated it.